src/analysis/monotone-analyzer-impl.h

```diff
@@ -0,0 +1,150 @@
+#ifndef wasm_analysis_monotone_analyzer_impl_h
+#define wasm_analysis_monotone_analyzer_impl_h
+
+#include <iostream>
+#include <unordered_map>
+
+#include "monotone-analyzer.h"
+
+namespace wasm::analysis {
+template<size_t N>
+BlockState<N>::BlockState(const BasicBlock* underlyingBlock)
+  : index(underlyingBlock->getIndex()), cfgBlock(underlyingBlock),
+    beginningState(BitsetPowersetLattice<N>::getBottom()),
+    endState(BitsetPowersetLattice<N>::getBottom()),
+    currState(BitsetPowersetLattice<N>::getBottom()) {}
+
+template<size_t N> void BlockState<N>::addPredecessor(BlockState* pred) {
+  predecessors.push_back(pred);
+}
+
+template<size_t N> void BlockState<N>::addSuccessor(BlockState* succ) {
+  successors.push_back(succ);
+}
+
+template<size_t N> BitsetPowersetLattice<N>& BlockState<N>::getFirstState() {
+  return beginningState;
+}
+
+template<size_t N> BitsetPowersetLattice<N>& BlockState<N>::getLastState() {
+  return endState;
+}
+
+// In our current limited implementation, we just update a new live variable
+// when it it is used in a get or set.
+template<size_t N> void BlockState<N>::visitLocalSet(LocalSet* curr) {
+  currState.value[curr->index] = true;
```

Comment:
Shouldn't this set the variable to `false`, since its previous value (the "after" state in the backward analysis) is not live?

Reply:
Fixed.